cmd/gateway/azure/gateway-azure.go

```diff
@@ -973,8 +1007,12 @@ func (a *azureObjects) PutObjectPart(ctx context.Context, bucket, object, upload
 		}
 
 		id := base64.StdEncoding.EncodeToString([]byte(minio.MustGetUUID()))
-		blob := a.client.GetContainerReference(bucket).GetBlobReference(object)
-		err = blob.PutBlockWithLength(id, uint64(subPartSize), io.LimitReader(data, subPartSize), nil)
+		blobURL := a.client.NewContainerURL(bucket).NewBlockBlobURL(object)
+		body, err := ioutil.ReadAll(io.LimitReader(data, subPartSize))
+		if err != nil {
+			return info, azureToObjectError(err, bucket, object)
+		}
+		_, err = blobURL.StageBlock(ctx, id, bytes.NewReader(body), azblob.LeaseAccessConditions{}, nil)
```

Comment:
Why we are not passing `io.LimitReader(data, subPartSize)` directly here, instead of storing 'data' of memory first and then recreating a reader ?

Reply:
Shouldn't we commit the blocks with its IDs as per - https://godoc.org/github.com/Azure/azure-storage-blob-go/azblob#BlockBlobURL.CommitBlockList @c-w

Reply:
@vadmeste This is similar to https://github.com/minio/minio/pull/8537#discussion_r349799346: the SDK requires `ReadSeeker` but we only have `Reader`.

@Praveenrajmani If I understood things correctly, the azure-storage-for-go SDK used [PutBlockList](https://godoc.org/github.com/Azure/azure-sdk-for-go/storage#Blob.PutBlockList) to transform a list of staged blocks into a blob and the equivalent of this API in the azure-storage-blob-go SDK is [CommitBlockList](https://godoc.org/github.com/Azure/azure-storage-blob-go/azblob#BlockBlobURL.CommitBlockList). If this is the case, the [previous implementation of PutObjectPart](https://github.com/minio/minio/blob/0a56e33ce1b50b79c8c67de6a8d57b33836cffe2/cmd/gateway/azure/gateway-azure.go#L952-L1008) doesn't seem to have called PutBlockList so the new implementation shouldn't call CommitBlockList either. What did I miss?

Reply:
@c-w I was referring to the complete multipart. Checked now, we are already doing that :+1: